pseudopotential_calculator/calculations/slab.py

```diff
@@ -119,11 +126,56 @@ def plot_energy_against_n_vacuum_layer(
         atom = get_calculator_atom(calculator)
 
         energies.append(get_atom_potential_energy(atom))
-        n_vacuum_layers = get_n_vacuum_layers_from_out_put(atom)
+        n_vacuum_layers = _get_n_vacuum_layers_from_slab(atom)
         n_vacuum_layer.append(n_vacuum_layers)
 
     return plot_data_comparison(
         ("Vacuum Height", np.array(n_vacuum_layer), "N layers"),
         ("Energy", np.array(energies), "J"),
         ax=ax,
     )
+
+
+def get_atom_displacement(atom: Atoms, n_th_layer: int) -> float:
+    heights = cast(list[float], atom.positions[:, 2])  # type: ignore bad lib
```

Comment:
this is not a list?

Reply:
This is what the error reports
```
Type of "positions" is partially unknown
Type of "positions" is "Any | ndarray[Any, dtype[bool_]] | Unknown"
```

Reply:
Yeah so cast it to an ndarray of np.dtype[np.float_]